lib/metasploit/framework/login_scanner/mssql.rb

```diff
@@ -52,8 +52,13 @@ def attempt_login(credential)
             else
               result_options[:status] = Metasploit::Model::Login::Status::INCORRECT
             end
-          rescue ::Rex::ConnectionError
+          rescue ::Rex::ConnectionError => e
             result_options[:status] = Metasploit::Model::Login::Status::UNABLE_TO_CONNECT
+            result_options[:proof] = e
+          rescue => e
+            elog(e)
+            result_options[:status] = Metasploit::Model::Login::Status::UNABLE_TO_CONNECT
+            result_options[:proof] = e
```

Comment:
Just wondering about taking the exception object and placing it in `[:proof]` which typically takes a string right?

Reply:
> Just wondering about taking the exception object and placing it in [:proof] which typically takes a string right?

Looks like we use a mix of `[:proof] = e` and `[:proof] = e.message` in the codebase, I don't believe this will cause an issue :+1:

https://github.com/rapid7/metasploit-framework/blob/a5aeb0082b68a8b37a9c0f867890b3ae56ded2b8/lib/metasploit/framework/login_scanner/smb.rb#L81-L90


https://github.com/rapid7/metasploit-framework/blob/a5aeb0082b68a8b37a9c0f867890b3ae56ded2b8/lib/metasploit/framework/login_scanner/mqtt.rb#L73-L77

With regards to the `invalidate_login` discussion, I'd like to keep that out of this PR for now - as it's a larger metasploit-framework issue IMO. When I've used Metasploit's login scanners with the database connected, it's been extremely slow relative to having it disabled